design-documents/graph-ql/coverage/reward-points.graphqls

```diff
@@ -0,0 +1,70 @@
+type Customer {
+    reward_points: RewardPoints @doc(description: "Customer reward points details")
+}
+
+type Cart {
+    applied_reward_points: RewardPointsAmount @doc(description: "Reward points ammount applied to the cart")
+}
+
+type Mutation {
+    applyRewardPointsToCart(cartId: ID!): ApplyRewardPointsToCartOutput @doc(description: "Apply all available points up to the cart total. Partial redemption is not available")
+    removeRewardPointsFromCart(cartId: ID!): RemoveRewardPointsFromCartOutput @doc(description: "Cancel usage of the previously applied reward points to cart")
+}
+
+type ApplyRewardPointsToCartOutput {
+    cart: Cart
+}
+
+type RemoveRewardPointsFromCartOutput {
+    cart: Cart
+}
+
+type RewardPoints {
+    balance: RewardPointsAmount @doc(description: "Current reward points balance")
+    exchange_rates: RewardPointsExchangeRates @doc(description: "Current reward points exchange rates")
+    subscription_status: RewardPointsSubscriptionStatus @doc(description: "Reward points related email subscription status")
+    balance_history: [RewardPointsBalanceHistoryItem] @doc(description: "Reward points balance history. This field will be set to null if disabled in the admin")
+}
+
+type RewardPointsAmount {
+    points: Float! @doc(description: "Reward points amount in points")
+    money: Money! @doc(description: "Reward points amount in store currency")
+}
+
+type RewardPointsExchangeRates @doc (description: "Exchange rates depend on the customer group"){
+    earning: Float! @doc(description: "How many points are earned per 1 currency spent")
+    redemption: Float! @doc(description: "How many points need to be redeemed to get 1 currency discount at the checkout")
+}
+
+type RewardPointsSubscriptionStatus {
+    balance_updates: SubscriptionStatus! @doc(description: "Customer subscription status to 'Reward points balance updates' emails")
```

Comment:
`balance_updates` seems a bit generic here, while the return type and the description give a lot more context.

Can we name this field in a way that it's less likely folks need to reference the docs? Maybe `balance_update_email_subscription`? Going to be a trade-off with verbosity, maybe we can find a happy middle ground.

Reply:
The type name already contains "SubscriptionStatus", so if you go from there it is "balance updates subscription status". Let me know if you still believe it is better to change it.

Reply:
That's a good point - I think it's good as-is 👍